src/gpu/intel/sycl/l0/utils.cpp

```diff
@@ -272,6 +287,115 @@ bool compare_ze_devices(const ::sycl::device &lhs, const ::sycl::device &rhs) {
     return lhs_ze_handle == rhs_ze_handle;
 }
 
+status_t get_device_ip(ze_device_handle_t device, uint32_t &ip_version) {
+    ze_device_properties_t deviceProps = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+    ze_device_ip_version_ext_t devicePropsIP
+            = {ZE_STRUCTURE_TYPE_DEVICE_IP_VERSION_EXT};
+    deviceProps.pNext = &devicePropsIP;
+    func_zeDeviceGetProperties(device, &deviceProps);
+    ip_version = devicePropsIP.ipVersion;
+    return status::success;
+}
+
+status_t get_lz_device_enabled_systolic_intel(
+        ze_device_handle_t device, bool &mayiusesystolic) {
+    ze_device_module_properties_t deviceModProps
+            = {ZE_STRUCTURE_TYPE_DEVICE_MODULE_PROPERTIES};
+    // Note: supported by Intel Driver 24.05 and onwards
+    ze_intel_device_module_dp_exp_properties_t deviceModPropsExt
+            = {ZE_STRUCTURE_INTEL_DEVICE_MODULE_DP_EXP_PROPERTIES};
+    deviceModProps.pNext = &deviceModPropsExt;
+
+    func_zeDeviceGetModuleProperties(device, &deviceModProps);
```

Comment:
Status check?

Reply:
added status check and status propagation. Thanks.